boot/bootutil/src/bootutil_priv.h

```diff
@@ -252,6 +252,41 @@ int boot_write_enc_key(const struct flash_area *fap, uint8_t slot,
 int boot_read_enc_key(int image_index, uint8_t slot, uint8_t *enckey);
 #endif
 
+/**
+ * Safe (non-overflowing) uint32_t addition.  Returns true, and stores
+ * the result in *dest if it can be done without overflow.  Otherwise,
+ * returns false.
+ */
+static inline bool boot_u32_safe_add(uint32_t *dest, uint32_t a, uint32_t b)
```

Comment:
The only thing is that their definitions are quite different, just by nature of the CPU being 32-bit. I'm not sure we're going to end up with more than two of these functions. But if we have more, it might be worth generating them.